central/networkgraph/flow/datastore/internal/store/postgres/store.go

```diff
@@ -545,18 +550,18 @@ func (s *flowStoreImpl) retryableRemoveMatchingFlows(ctx context.Context, keyMat
 	}
 	defer release()
 
+	rows, err := conn.Query(ctx, walkStmt, s.clusterID)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
 	// TODO(ROX-9921) Look at refactoring how these predicates work as an overall refactor of flows.
 	// This operation matches if the either the dest or src deployment no longer exists AND then
 	// if the last seen time is outside a time window.  Since we do not yet know what deployments exist
 	// in Postgres we cannot fully do this work in SQL.  Additionally, there may be issues with the synchronization
 	// of when flow is created vs a deployment deleted that may also make that problematic.
 	if keyMatchFn != nil {
-		rows, err := conn.Query(ctx, walkStmt, s.clusterID)
```

Comment:
Actually, this seems like now we've made the query have greater scope? whereas it wasn't running before?

Reply:
yeah you are right.  Also did that before I understood what Next was doing.  I will move it back.